### DISCUSSION TITLE
Add ability to hide all visible difficulties

### DISCUSSION THREAD

**Discussion by gowon11** [ppy/osu#30644] [Ideas]
Since lazer uses a different method of storage for beatmaps we can't directly access the maps folder to delete "useless" files and even entire mapsets to save up some space. Not only that but it seems like the delete all visible maps button isn't present in lazer at all. This raises a concern for me in regards to the amount of storage that lazer might take from my drive since theres no middle ground in terms of map deletion. On stable a routine of mine was downloading some of the beatmap packs from the website and then deleting maps i didn't played from my song folder directly, then i ran a tool to delete unused files from the songs folder, such as [TechNobo osu!cleaner](https://github.com/TCNOco/TcNo-osu-Cleaner). On lazer however, the only options given by the game to delete maps is is delete all maps, delete all difficulties of the current select map and delete video files.
![osu!_mMWUZNXPnT](https://github.com/user-attachments/assets/f618be17-d536-4f17-bdbc-add5791a26bf)![osu_2024-11-15_08-05-46](https://github.com/user-attachments/assets/975d9dd6-bacb-4e74-8fa8-5c951310c9bc)
While its nice to delete videos, the maps delete option doesn't solve the problem of storage. I don't want to delete all my collection of maps that i imported from stable, and personally i don't feel like deleting maps one by one by hand.

My request here would be to bring back the option to delete all visible maps, but properly done this time. Those of you who ever tried to run this on stable know that all it did was crash the game, and even when it worked all it did was delete the .osu file on the song folder, it didn't delete the remaining files that actually took space like the song file, background, etc.

Since lazer is so well built i was wondering if it could be implemented in a way that doesn't just crash the game when it tried to delete more than 300 maps at once, and that it **actually** delete the map fully. This would solve the problem i am facing, where my installation is getting increasingly bigger with no way to delete garbage to save some storage.

**Comment by gowon11** [comment]
tl:dr - add more ways to manage osu storage in game as external tools can't be used anymore

**Comment by peppy** [comment]
Deleting .osu files will not be an option as they are tiny. We offer the ability to hide. We may add the ability to "hide all visible difficulties" which would cover your issue, I believe.

**Reply by gowon11** [reply]
That seems to cover the part about removing unwanted maps but i would like to emphasize the topic of storage usage of the game.

**Reply by peppy** [reply]
Please re-read my comment. The first half addresses your concerns.

**Reply by gowon11** [reply]
That doesn't really addresses my concerns because on my own post i also mention that i don't want .osu files to be deleted for the same reasons that you mentioned.
> and even when it worked all it did was delete the .osu file on the song folder, it didn't delete the remaining files that actually took space like the song file, background, etc.

The request here was about these other files: beatmap skin files, beatmap hitsound, storyboard stuff. And also ways to delete mass entire mapsets, including yes the .osu file but also all the remaining files of said beatmap, but without needing to start over by just deleting every map installed, which is the option given in the settings.
